Julia/nsw_experimental.jl

```diff
@@ -0,0 +1,51 @@
+using Chain, 
+      DataFrames, 
+      FileIO, 
+      HTTP, 
+      Statistics
+
+function read_data(df) 
+    path = "https://raw.github.com/scunning1975/mixtape/master/" * df
+    @chain path begin
+        HTTP.download
+        load
+        DataFrame
+    end
+end
+
+nsw_dw = read_data("nsw_mixtape.dta")
+
+# careful: subset instead of subset! to not replace the data
+
+@chain nsw_dw begin
+    subset(_, :treat => ByRow(==(1)))
+    describe(_.re78)
+end
+
+mean1 = @chain nsw_dw begin
+    subset(_, :treat => ByRow(==(1)))
+    mean(_.re78)
+end
+
+insertcols!(nsw_dw, :y1 => mean1)
+
+@chain nsw_dw begin
+    subset(_, :treat => ByRow(==(0)))
+    describe(_.re78)
+end
+
+mean0 = @chain nsw_dw begin
+    subset(_, :treat => ByRow(==(0)))
+    mean(_.re78)
+end
+
+insertcols!(nsw_dw, :y0 => mean0)
+
+ate = unique(nsw_dw.y1 - nsw_dw.y0)
```

Comment:
I feel like this would be easier:

```ate = @chain nsw_dw begin
    groupby(:treat)
    combine(:re78 => mean => :y)
    diff(_.y)
    only
end
```

The `only` is optional but converts single element vectors to scalars